README.md

```diff
@@ -34,7 +34,7 @@ Below is a basic client example. See the `examples/` directory for more.
 ```rust
 fn fetch_inbox_top() -> imap::error::Result<Option<String>> {
 
-    let client = imap::ClientBuilder::new("imap.example.com", 993).native_tls()?;
+    let client = imap::ClientBuilder::new("imap.example.com", 993).tls().connect()?;
```

Comment:
I think we should keep the `native_tls` and `rustls` constructors for convenience.

Reply:
Also, I think those constructors could utilize `connect_tls_with` to avoid the `Box<dyn>` (and thus get a slight performance boost).

Reply:
@jonhoo which is why I kinda like the "enum" approach that the Ldap crate used as it removed the need for the box.

And yes, I can add back in the nativeTls / rusttls methods.

Reply:
When you do, I think we can also avoid changing all these examples

Reply:
well, so, I would think we'd want the examples to be in the preferable "non-surprising" syntax. (e.g. it doesn't break if they switch backends)  (which is why I opted for dropping those methods to begin with)

Also, if you are concerned about performance of the Box we can use an enum of 
```rust
enum Connection {
    Tcp(TcpStream),
    NativeTls(NativeTlsStream),
    RustTls(RusTlsStream),
    Custom(Box<ImapConnection>),
}
```

Reply:
Hmm, maybe. I'm torn, because it's _also_ useful to show folks how they can choose a particular TLS backend to use. I'll go whichever way you prefer here :+1: 

No, `Box<dyn>` is generally going to be better performance than dispatch-through-enum I think.

Reply:
We can add examples showing the variations, and those two methods (rustls, and native_tls) that I added back in have doc-comment examples showing usage.